implementations/micrometer-registry-stackdriver/src/test/java/io/micrometer/stackdriver/StackdriverIntegrationTest.java

```diff
@@ -0,0 +1,152 @@
+            .isEmpty();
+    }
+
+    @Test
+    void timerWithSlos() {
+        Timer slos = Timer.builder("micrometer.stackdriver.it.timer.slo")
+            .maximumExpectedValue(Duration.ofMillis(2))
+            .serviceLevelObjectives(Duration.ofMillis(1), Duration.ofMillis(2))
+            .register(registry);
+        slos.record(Duration.ofMillis(1));
+        slos.record(Duration.ofMillis(2));
+        slos.record(Duration.ofMillis(3));
+
+        System.out.println(slos.getId().toString() + batch.distribution(slos.takeSnapshot(), true));
+        publishAndVerify();
+    }
+
+    @Test
+    void timerWithPercentileHistogram() {
+        Timer percentileHistogram = Timer.builder("micrometer.stackdriver.it.timer.percentilehistogram")
+            .publishPercentileHistogram()
+            .register(registry);
+
+        percentileHistogram.record(Duration.ofMillis(1));
+        percentileHistogram.record(Duration.ofMillis(2));
+        percentileHistogram.record(Duration.ofMillis(3));
+
+        System.out.println(
+                percentileHistogram.getId().toString() + batch.distribution(percentileHistogram.takeSnapshot(), true));
+        publishAndVerify();
+    }
+
+    @Test
+    void timerWithClientPercentiles() {
+        Timer clientPercentiles = Timer.builder("micrometer.stackdriver.it.timer.clientpercentiles")
+            .publishPercentiles(0.5, 0.99)
+            .register(registry);
+        clientPercentiles.record(Duration.ofMillis(1));
+        clientPercentiles.record(Duration.ofMillis(2));
+        clientPercentiles.record(Duration.ofMillis(3));
+
+        System.out
+            .println(clientPercentiles.getId().toString() + batch.distribution(clientPercentiles.takeSnapshot(), true));
+        publishAndVerify();
+    }
+
+    @Test
+    void timerWithClientPercentilesAndSlo() {
+        Timer clientPercentilesAndSlos = Timer.builder("micrometer.stackdriver.it.timer.clientpercentilesandslo")
+            .serviceLevelObjectives(Duration.ofMillis(2))
+            .publishPercentiles(0.5, 0.99)
+            .register(registry);
+        clientPercentilesAndSlos.record(Duration.ofMillis(1));
+        clientPercentilesAndSlos.record(Duration.ofMillis(2));
+        clientPercentilesAndSlos.record(Duration.ofMillis(3));
+
+        System.out.println(clientPercentilesAndSlos.getId().toString()
+                + batch.distribution(clientPercentilesAndSlos.takeSnapshot(), true));
+        publishAndVerify();
+    }
+
+    @Test
+    void summary() {
+        DistributionSummary summary = DistributionSummary.builder("micrometer.stackdriver.it.summary")
+            .register(registry);
+        summary.record(10d);
+        summary.record(20d);
+        summary.record(30d);
+
+        System.out.println(summary.getId().toString() + batch.distribution(summary.takeSnapshot(), false));
+        publishAndVerify();
+    }
+
+    @Test
+    void summaryWithOneSlo() {
```

Comment:
Should we mirror the Timer tests for DistributionSummary too?

Reply:
We could, but other than unit and conversion of that unit, there isn't really a difference between the two when it comes to the logic or output. I was trying to make sure we had coverage of all the different underlying histograms and edge cases in the logic for producing the Stackdriver Distribution. If I've missed something, we should definitely add it. Otherwise I'm not sure we need to do the same configuration with each (famous last words).